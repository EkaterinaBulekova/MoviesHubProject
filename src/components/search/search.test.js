import React from 'react';
import { mount } from 'enzyme';
import Search from './search';
import toJson from 'enzyme-to-json';

describe('<Search />', () => {
  it('should render as expect', () => {
    const testOnClick = jest.fn(); 
    const component = mount(<Search onSearchClick={testOnClick}/>);
    const searchButton = component.find('button.search-button');
    const filterButtons = component.find('button.search-filter-button');
    const input = component.find('input');
    component.instance().setState = jest.fn();
    component.update();

    expect(toJson(component)).toMatchSnapshot();

    expect(component.find('div.search-field-title').text()).toBe('FIND YOUR MOVIE');
    
    expect(input.hasClass('search-field-input'));
    input.simulate('keyUp',{key:'Enter'});
    expect(testOnClick.mock.calls.length).toBe(1);
    expect(testOnClick).toBeCalled();
    input.simulate('change',{value:'value'});
    expect(component.instance().setState).toBeCalled();
    //expect(component.instance().setState).toBeCalledWith({search:'value'});

    searchButton.simulate('click');
    expect(testOnClick.mock.calls.length).toBe(2);
    expect(testOnClick).toBeCalled();

    expect(component.find('div.search-filter-title').text()).toBe('SEARCH BY');
    expect(filterButtons).toHaveLength(2);
    filterButtons.at('1').simulate('click');
    expect(component.instance().setState).toBeCalled();
    expect(component.instance().setState).toBeCalledWith({searchBy:'genres'});
    //expect(component.state().searchBy).toBe('genres');
    filterButtons.at('0').simulate('click');

    component.unmount();
  });
});