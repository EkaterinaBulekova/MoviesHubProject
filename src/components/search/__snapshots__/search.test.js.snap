// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> should render as expect 1`] = `
<Search
  onSearchClick={[MockFunction]}
>
  <form
    className="search-component"
    onSubmit={[Function]}
  >
    <CustomInput
      className="search-field"
      title="FIND YOUR MOVIE"
    >
      <div
        className="search-field"
      >
        <div
          className="search-field-title"
        >
          FIND YOUR MOVIE
        </div>
        <input
          className="search-field-input"
        />
      </div>
    </CustomInput>
    <SearchFilter
      onClick={[Function]}
      searchBy="title"
    >
      <ButtonsGroup
        active="title"
        group={
          Object {
            "buttons": Array [
              Object {
                "name": "TITLE",
                "result": "title",
              },
              Object {
                "name": "GENRE",
                "result": "genres",
              },
            ],
            "className": "search-filter",
            "title": "SEARCH BY",
          }
        }
        onClick={[Function]}
      >
        <div
          className="search-filter"
        >
          <div
            className="search-filter-title"
          >
            SEARCH BY
          </div>
          <Button
            className="search-filter-button active"
            key="title"
            name="TITLE"
            onClick={[Function]}
          >
            <button
              className="btn search-filter-button active"
              name="TITLE"
              onClick={[Function]}
              type="button"
            >
              TITLE
            </button>
          </Button>
          <Button
            className="search-filter-button"
            key="genres"
            name="GENRE"
            onClick={[Function]}
          >
            <button
              className="btn search-filter-button"
              name="GENRE"
              onClick={[Function]}
              type="button"
            >
              GENRE
            </button>
          </Button>
        </div>
      </ButtonsGroup>
    </SearchFilter>
    <Button
      className="search-button active"
      name="SEARCH"
      type="submit"
    >
      <button
        className="btn search-button active"
        name="SEARCH"
        type="submit"
      >
        SEARCH
      </button>
    </Button>
  </form>
</Search>
`;
